File: listar_productos.py
from flask import Flask, jsonify
import mysql.connector

app = Flask(__name__)

# Configuración de la base de datos
db_config = {
    'host': 'localhost',
    'user': 'root',
    'password': 'C#L#ST#ybl4nco',
    'database': 'bd_unificada'
}

@app.route('/listar_productos', methods=['GET'])
def listar_productos():
    try:
        # Conexión a la base de datos
        conn = mysql.connector.connect(**db_config)
        cursor = conn.cursor(dictionary=True)

        # Consulta para obtener los primeros 5 productos
        cursor.execute("SELECT * FROM producto LIMIT 5")
        productos = cursor.fetchall()

        cursor.close()
        conn.close()

        return jsonify(productos), 200

    except mysql.connector.Error as err:
        return jsonify({'error': str(err)}), 500

# Punto de entrada
if __name__ == '__main__':
    app.run(debug=True, port=5000)

File: node/listar_productos.js
const express = require('express');
const mysql = require('mysql2');
const app = express();

// Configuración de la base de datos
const db = mysql.createConnection({
  host: 'localhost',
  user: 'root',
  password: 'root',
  database: 'bd_unificada'
});

// Conectar a la base de datos
db.connect((err) => {
  if (err) {
    console.error('Error al conectar a la base de datos:', err.message);
    process.exit(1);
  }
  console.log('Conectado a la base de datos MySQL');
});

// Ruta GET para listar los primeros 5 productos
app.get('/listar_productos', (req, res) => {
  const query = 'SELECT * FROM producto LIMIT 5';

  db.query(query, (err, results) => {
    if (err) {
      console.error('Error al consultar productos:', err.message);
      return res.status(500).json({ error: err.message });
    }

    res.status(200).json(results);
  });
});

// Iniciar servidor
app.listen(5000, () => {
  console.log('Microservicio "listar_productos" escuchando en http://localhost:5000');
});
